File: alice/paskills/dialogovo/common/src/main/java/domain/DeveloperType.java
package ru.yandex.alice.paskill.dialogovo.domain;

import javax.annotation.Nullable;

import ru.yandex.alice.megamind.protos.analytics.scenarios.dialogovo.Dialogovo.TSkillDeveloperType;

public enum DeveloperType {
    Yandex("yandex"),
    External("external");

    private final String value;

    DeveloperType(String value) {
        this.value = value;
    }

    public static DeveloperType fromString(@Nullable String developerType) {
        if ("yandex".equals(developerType)) {
            return DeveloperType.Yandex;
        } else {
            return DeveloperType.External;
        }
    }

    public String getValue() {
        return value;
    }

    public static TSkillDeveloperType getProtoDeveloperType(DeveloperType source) {
        switch (source) {
            case Yandex:
                return TSkillDeveloperType.Yandex;
            default:
                return TSkillDeveloperType.External;
        }
    }
}

File: alice/kronstadt/scenarios/theremin/src/main/java/DeveloperType.kt
package ru.yandex.alice.paskill.dialogovo.scenarios.theremin

import ru.yandex.alice.megamind.protos.analytics.scenarios.dialogovo.Dialogovo.TSkillDeveloperType
import javax.annotation.Nullable

enum class DeveloperType(val value: String) {
    Yandex("yandex"),
    External("external");

    companion object {
        fun fromString(@Nullable developerType: String?): DeveloperType {
            return if ("yandex".equals(developerType)) {
                Yandex
            } else {
                External
            }
        }

        fun getProtoDeveloperType(source: DeveloperType): TSkillDeveloperType {
            return when (source) {
                Yandex -> TSkillDeveloperType.Yandex
                else -> TSkillDeveloperType.External
            }
        }
    }
}
